src/containers/BuildUnsignedSweepCoin/BuildUnsignedSweepCoin.tsx

```diff
@@ -12,6 +20,33 @@ import { LitecoinForm } from './LitecoinForm';
 import { RippleForm } from './RippleForm';
 import { TronForm } from './TronForm';
 
+async function deriveKeyWithSeedAndToken(
+  token: string | undefined,
+  ...params: Parameters<typeof window.queries.deriveKeyWithSeed>
+) {
+  const [coin, ...rest] = params;
+  let keyAndDerivationPath;
+  if (token) {
+    try {
+      keyAndDerivationPath = await window.queries.deriveKeyWithSeed(
+        token,
+        ...rest
+      );
+    } catch (e) {
+      keyAndDerivationPath = await window.queries.deriveKeyWithSeed(
+        coin,
+        ...rest
+      );
+    }
+  } else {
+    keyAndDerivationPath = await window.queries.deriveKeyWithSeed(
+      coin,
+      ...rest
+    );
+  }
+  return keyAndDerivationPath;
+}
+
 async function includePubsFor<
```

Comment:
Don't do it like this, revert it back then create a new function which calls it with a token or a coin